src/seedu/addressbook/data/person/Name.java

```diff
@@ -59,5 +59,13 @@ public boolean equals(Object other) {
     public int hashCode() {
         return fullName.hashCode();
     }
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if ...
+     */
+     public boolean isSimilar(Name other) {
```

Comment:
instead of writing the method first, you should write the test. obviously it will fail (because it can't compile), but writing tests that initially fail then writing minimal amount of code to pass the test is part of the TDD process.

![](https://i.redd.it/671bo01674ey.jpg)